Official build fails due to missing windows_arm job

Official builds have started failing with:

```
workloadsbuild depends on unknown job installer__coreclr__windows_arm_Release_.
```

See for example https://dev.azure.com/dnceng/internal/_build/results?buildId=2178029&view=results. I believe caused by https://github.com/dotnet/runtime/pull/85947. @BruceForstall

Probably we need to at least remove the arm dependencies from
https://github.com/dotnet/runtime/blob/11e22cebdf2e76795be088d12830eed767367d51/eng/pipelines/runtime-official.yml#L527 and
https://github.com/dotnet/runtime/blob/11e22cebdf2e76795be088d12830eed767367d51/eng/pipelines/mono/templates/workloads-build.yml#L75

Looks like the official build is failing in the `Build crossdac publishing` step:

```
D:\a\_work\1\s\.packages\microsoft.dotnet.build.tasks.packaging\8.0.0-beta.23259.5\build\Packaging.targets(1264,5): error : Error when creating nuget lib package from D:\a\_work\1\s\artifacts\packages\Release\specs\runtime.linux-musl-arm.Microsoft.CrossOsDiag.Private.CoreCLR.nuspec. NuGet.Packaging.Core.PackagingException: File not found: 'D:\a\_work\1\s\artifacts\CoreCLRCrossDacArtifacts\CoreCLRCrossDacArtifacts\linux_musl.arm.Release\x86\mscordaccore.pdb'. [D:\a\_work\1\s\src\coreclr\.nuget\Microsoft.CrossOsDiag.Private.CoreCLR\Microsoft.CrossOsDiag.Private.CoreCLR.pkgproj]
D:\a\_work\1\s\.packages\microsoft.dotnet.build.tasks.packaging\8.0.0-beta.23259.5\build\Packaging.targets(1264,5): error :    at NuGet.Packaging.PackageBuilder.AddFiles(String basePath, String source, String destination, String exclude) [D:\a\_work\1\s\src\coreclr\.nuget\Microsoft.CrossOsDiag.Private.CoreCLR\Microsoft.CrossOsDiag.Private.CoreCLR.pkgproj]
D:\a\_work\1\s\.packages\microsoft.dotnet.build.tasks.packaging\8.0.0-beta.23259.5\build\Packaging.targets(1264,5): error :    at NuGet.Packaging.PackageBuilder.PopulateFiles(String basePath, IEnumerable`1 files) [D:\a\_work\1\s\src\coreclr\.nuget\Microsoft.CrossOsDiag.Private.CoreCLR\Microsoft.CrossOsDiag.Private.CoreCLR.pkgproj]
D:\a\_work\1\s\.packages\microsoft.dotnet.build.tasks.packaging\8.0.0-beta.23259.5\build\Packaging.targets(1264,5): error :    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.Pack(String nuspecPath, String nupkgPath, Manifest manifest, Boolean packSymbols) in /_/src/Microsoft.DotNet.Build.Tasks.Packaging/src/NuGetPack.cs:line 261 [D:\a\_work\1\s\src\coreclr\.nuget\Microsoft.CrossOsDiag.Private.CoreCLR\Microsoft.CrossOsDiag.Private.CoreCLR.pkgproj]
D:\a\_work\1\s\.packages\microsoft.dotnet.build.tasks.packaging\8.0.0-beta.23259.5\build\Packaging.targets(1264,5): error : Error when creating nuget lib package from D:\a\_work\1\s\artifacts\packages\Release\specs\runtime.linux-arm.Microsoft.CrossOsDiag.Private.CoreCLR.nuspec. NuGet.Packaging.Core.PackagingException: File not found: 'D:\a\_work\1\s\artifacts\CoreCLRCrossDacArtifacts\CoreCLRCrossDacArtifacts\linux.arm.Release\x86\mscordaccore.pdb'. [D:\a\_work\1\s\src\coreclr\.nuget\Microsoft.CrossOsDiag.Private.CoreCLR\Microsoft.CrossOsDiag.Private.CoreCLR.pkgproj]
D:\a\_work\1\s\.packages\microsoft.dotnet.build.tasks.packaging\8.0.0-beta.23259.5\build\Packaging.targets(1264,5): error :    at NuGet.Packaging.PackageBuilder.AddFiles(String basePath, String source, String destination, String exclude) [D:\a\_work\1\s\src\coreclr\.nuget\Microsoft.CrossOsDiag.Private.CoreCLR\Microsoft.CrossOsDiag.Private.CoreCLR.pkgproj]
D:\a\_work\1\s\.packages\microsoft.dotnet.build.tasks.packaging\8.0.0-beta.23259.5\build\Packaging.targets(1264,5): error :    at NuGet.Packaging.PackageBuilder.PopulateFiles(String basePath, IEnumerable`1 files) [D:\a\_work\1\s\src\coreclr\.nuget\Microsoft.CrossOsDiag.Private.CoreCLR\Microsoft.CrossOsDiag.Private.CoreCLR.pkgproj]
D:\a\_work\1\s\.packages\microsoft.dotnet.build.tasks.packaging\8.0.0-beta.23259.5\build\Packaging.targets(1264,5): error :    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.Pack(String nuspecPath, String nupkgPath, Manifest manifest, Boolean packSymbols) in /_/src/Microsoft.DotNet.Build.Tasks.Packaging/src/NuGetPack.cs:line 261 [D:\a\_work\1\s\src\coreclr\.nuget\Microsoft.CrossOsDiag.Private.CoreCLR\Microsoft.CrossOsDiag.Private.CoreCLR.pkgproj]
##[error].packages\microsoft.dotnet.build.tasks.packaging\8.0.0-beta.23259.5\build\Packaging.targets(1264,5): error : (NETCORE_ENGINEERING_TELEMETRY=Build) Error when creating nuget lib package from D:\a\_work\1\s\artifacts\packages\Release\specs\runtime.linux-musl-arm.Microsoft.CrossOsDiag.Private.CoreCLR.nuspec. NuGet.Packaging.Core.PackagingException: File not found: 'D:\a\_work\1\s\artifacts\CoreCLRCrossDacArtifacts\CoreCLRCrossDacArtifacts\linux_musl.arm.Release\x86\mscordaccore.pdb'.
```

@sbomer @jkotas @jkoritzinsky Anyone know how this works? @jkotas Maybe the linux-arm cross-dac build was inadvertently removed?

--- a/eng/pipelines/coreclr/templates/build-job.yml
+++ b/eng/pipelines/coreclr/templates/build-job.yml
@@ -202,6 +202,10 @@ jobs:
       - script: $(Build.SourcesDirectory)/src/coreclr/build-runtime$(scriptExt) $(buildConfig) $(archType) -hostarch x64 $(osArg) -ci $(compilerArg) -component crosscomponents -cmakeargs "-DCLR_CROSS_COMPONENTS_BUILD=1" $(officialBuildIdArg) $(clrRuntimePortableBuildArg)
         displayName: Build CoreCLR Cross-Arch Tools (Tools that run on x64 targeting x86)
 
+    - ${{ if and(eq(parameters.osGroup, 'windows'), eq(parameters.archType, 'x86')) }}:
+      - script: $(Build.SourcesDirectory)/src/coreclr/build-runtime$(scriptExt) $(buildConfig) arm -hostarch x86 $(osArg) -ci $(compilerArg) -component crosscomponents -cmakeargs "-DCLR_CROSS_COMPONENTS_BUILD=1" $(officialBuildIdArg) $(clrRuntimePortableBuildArg)
+        displayName: Build CoreCLR Cross-Arch Tools (Tools that run on win-x86 targeting arm)
+
     - ${{ if in(parameters.osGroup, 'osx', 'ios', 'tvos') }}:
       - script: |
           du -sh $(Build.SourcesDirectory)/*

--- a/eng/pipelines/coreclr/templates/build-job.yml
+++ b/eng/pipelines/coreclr/templates/build-job.yml
@@ -287,7 +287,7 @@ jobs:
             ${{ else }}:
               hostArchType: x64
 
-    - ${{ if and(in(parameters.osGroup, 'windows'), eq(parameters.archType, 'x86'), ne(parameters.compilerName, 'gcc'), ne(parameters.testGroup, 'clrTools'), eq(parameters.pgoType, '')) }}:
+    - ${{ if and(in(parameters.osGroup, 'windows'), eq(parameters.archType, 'x86'), ne(parameters.testGroup, 'clrTools'), eq(parameters.pgoType, '')) }}:
         - template: /eng/pipelines/coreclr/templates/crossdac-build.yml
           parameters:
             archType: arm

--- a/eng/pipelines/coreclr/templates/build-job.yml
+++ b/eng/pipelines/coreclr/templates/build-job.yml
@@ -202,10 +202,6 @@ jobs:
       - script: $(Build.SourcesDirectory)/src/coreclr/build-runtime$(scriptExt) $(buildConfig) $(archType) -hostarch x64 $(osArg) -ci $(compilerArg) -component crosscomponents -cmakeargs "-DCLR_CROSS_COMPONENTS_BUILD=1" $(officialBuildIdArg) $(clrRuntimePortableBuildArg)
         displayName: Build CoreCLR Cross-Arch Tools (Tools that run on x64 targeting x86)
 
-    - ${{ if and(eq(parameters.osGroup, 'windows'), eq(parameters.archType, 'x86')) }}:
-      - script: $(Build.SourcesDirectory)/src/coreclr/build-runtime$(scriptExt) $(buildConfig) arm -hostarch x86 $(osArg) -ci $(compilerArg) -component crosscomponents -cmakeargs "-DCLR_CROSS_COMPONENTS_BUILD=1" $(officialBuildIdArg) $(clrRuntimePortableBuildArg)
-        displayName: Build CoreCLR Cross-Arch Tools (Tools that run on win-x86 targeting arm)
-
     - ${{ if in(parameters.osGroup, 'osx', 'ios', 'tvos') }}:
       - script: |
           du -sh $(Build.SourcesDirectory)/*
@@ -290,6 +286,15 @@ jobs:
               hostArchType: x86
             ${{ else }}:
               hostArchType: x64
+
+    - ${{ if and(in(parameters.osGroup, 'windows'), eq(parameters.archType, 'x86'), ne(parameters.compilerName, 'gcc'), ne(parameters.testGroup, 'clrTools'), eq(parameters.pgoType, '')) }}:
+        - template: /eng/pipelines/coreclr/templates/crossdac-build.yml
+          parameters:
+            archType: arm
+            osGroup: ${{ parameters.osGroup }}
+            osSubgroup: ${{ parameters.osSubgroup }}
+            isOfficialBuild: ${{ parameters.signBinaries }}
+            hostArchType: x86
 
     - ${{ if and(ne(parameters.compilerName, 'gcc'), ne(parameters.testGroup, ''), ne(parameters.testGroup, 'clrTools'), ne(parameters.disableClrTest, true)) }}:
       # Publish test native components for consumption by test execution.

--- a/eng/pipelines/coreclr/templates/crossdac-build.yml
+++ b/eng/pipelines/coreclr/templates/crossdac-build.yml
@@ -56,7 +56,7 @@ steps:
         CopyAndVerifyCrossOsAssets -CrossDacDir $buildLinuxDacRootFolderPath -TargetDir $buildLinuxDacStagingPath
 
         Write-Host "Final directory contents:"
-        ls -R '$(crossDacArtifactPath)'
+        ls -R $(crossDacArtifactPath)
 
       displayName: Gather CrossDac Artifacts
 

--- a/eng/pipelines/coreclr/templates/crossdac-build.yml
+++ b/eng/pipelines/coreclr/templates/crossdac-build.yml
@@ -52,12 +52,11 @@ steps:
         $buildLinuxDacStagingPath = "$(crossDacArtifactPath)/Linux.${{ parameters.archType }}.$(buildConfigUpper)/${{ parameters.hostArchType }}"
         
         
-        CopyAndVerifyCrossOsAssets -CrossDacDir $buildMuslDacRootFolderPath -TargetDir '$buildMuslStagingPath'
-        CopyAndVerifyCrossOsAssets -CrossDacDir $buildLinuxDacRootFolderPath 
- -TargetDir '$buildLinuxDacStagingPath'
+        CopyAndVerifyCrossOsAssets -CrossDacDir '$buildMuslDacRootFolderPath' -TargetDir '$buildMuslStagingPath'
+        CopyAndVerifyCrossOsAssets -CrossDacDir '$buildLinuxDacRootFolderPath' -TargetDir '$buildLinuxDacStagingPath'
 
         Write-Host "Final directory contents:"
-        ls -R $(crossDacArtifactPath)
+        ls -R '$(crossDacArtifactPath)'
 
       displayName: Gather CrossDac Artifacts
 

--- a/eng/pipelines/coreclr/templates/crossdac-build.yml
+++ b/eng/pipelines/coreclr/templates/crossdac-build.yml
@@ -45,8 +45,15 @@ steps:
           $availableFiles | %{ cp $_.FullName $targetDir -v }
         }
 
-        CopyAndVerifyCrossOsAssets -CrossDacDir $(buildMuslDacRootFolderPath) -TargetDir '$(buildMuslDacStagingPath)/${{ parameters.hostArchType }}'
-        CopyAndVerifyCrossOsAssets -CrossDacDir $(buildLinuxDacRootFolderPath) -TargetDir '$(buildLinuxDacStagingPath)/${{ parameters.hostArchType }}'
+        $buildMuslDacRootFolderPath = "$(Build.SourcesDirectory)/artifacts/bin/coreclr/alpine.${{ parameters.archType }}.$(buildConfigUpper)"
+        $buildMuslStagingPath = "$(crossDacArtifactPath)/Linux_musl.${{ parameters.archType }}.$(buildConfigUpper)/${{ parameters.hostArchType }}"
+        
+        $buildLinuxDacRootFolderPath = "$(Build.SourcesDirectory)/artifacts/bin/coreclr/Linux.${{ parameters.archType }}.$(buildConfigUpper)"
+        $buildLinuxDacStagingPath = "$(crossDacArtifactPath)/Linux.${{ parameters.archType }}.$(buildConfigUpper)/${{ parameters.hostArchType }}"
+        
+        
+        CopyAndVerifyCrossOsAssets -CrossDacDir $(buildMuslDacRootFolderPath) -TargetDir '$buildMuslStagingPath'
+        CopyAndVerifyCrossOsAssets -CrossDacDir $(buildLinuxDacRootFolderPath) -TargetDir '$buildLinuxDacStagingPath'
 
         Write-Host "Final directory contents:"
         ls -R $(crossDacArtifactPath)

--- a/eng/pipelines/coreclr/templates/crossdac-build.yml
+++ b/eng/pipelines/coreclr/templates/crossdac-build.yml
@@ -52,8 +52,9 @@ steps:
         $buildLinuxDacStagingPath = "$(crossDacArtifactPath)/Linux.${{ parameters.archType }}.$(buildConfigUpper)/${{ parameters.hostArchType }}"
         
         
-        CopyAndVerifyCrossOsAssets -CrossDacDir $(buildMuslDacRootFolderPath) -TargetDir '$buildMuslStagingPath'
-        CopyAndVerifyCrossOsAssets -CrossDacDir $(buildLinuxDacRootFolderPath) -TargetDir '$buildLinuxDacStagingPath'
+        CopyAndVerifyCrossOsAssets -CrossDacDir $buildMuslDacRootFolderPath -TargetDir '$buildMuslStagingPath'
+        CopyAndVerifyCrossOsAssets -CrossDacDir $buildLinuxDacRootFolderPath 
+ -TargetDir '$buildLinuxDacStagingPath'
 
         Write-Host "Final directory contents:"
         ls -R $(crossDacArtifactPath)

--- a/eng/pipelines/coreclr/templates/crossdac-build.yml
+++ b/eng/pipelines/coreclr/templates/crossdac-build.yml
@@ -52,8 +52,8 @@ steps:
         $buildLinuxDacStagingPath = "$(crossDacArtifactPath)/Linux.${{ parameters.archType }}.$(buildConfigUpper)/${{ parameters.hostArchType }}"
         
         
-        CopyAndVerifyCrossOsAssets -CrossDacDir '$buildMuslDacRootFolderPath' -TargetDir '$buildMuslStagingPath'
-        CopyAndVerifyCrossOsAssets -CrossDacDir '$buildLinuxDacRootFolderPath' -TargetDir '$buildLinuxDacStagingPath'
+        CopyAndVerifyCrossOsAssets -CrossDacDir $buildMuslDacRootFolderPath -TargetDir $buildMuslStagingPath
+        CopyAndVerifyCrossOsAssets -CrossDacDir $buildLinuxDacRootFolderPath -TargetDir $buildLinuxDacStagingPath
 
         Write-Host "Final directory contents:"
         ls -R '$(crossDacArtifactPath)'

--- a/eng/pipelines/coreclr/templates/xplat-pipeline-job.yml
+++ b/eng/pipelines/coreclr/templates/xplat-pipeline-job.yml
@@ -76,18 +76,6 @@ jobs:
     - name: crossDacArtifactPath
       value: $(Build.SourcesDirectory)/artifacts/$(buildCrossDacArtifactName)
 
-    - name: buildMuslDacRootFolderPath
-      value: '$(Build.SourcesDirectory)/artifacts/bin/coreclr/alpine.$(archType).$(buildConfigUpper)'
-
-    - name: buildMuslDacStagingPath
-      value: '$(crossDacArtifactPath)/Linux_musl.$(archType).$(buildConfigUpper)'
-
-    - name: buildLinuxDacRootFolderPath
-      value: '$(Build.SourcesDirectory)/artifacts/bin/coreclr/Linux.$(archType).$(buildConfigUpper)'
-
-    - name: buildLinuxDacStagingPath
-      value: '$(crossDacArtifactPath)/Linux.$(archType).$(buildConfigUpper)'
-
     # We need this because both mono and coreclr build currently depends on CoreClr
     - name: coreClrProductArtifactName
       value: 'CoreCLRProduct_${{ parameters.pgoType }}_${{ parameters.runtimeVariant }}_$(osGroup)$(osSubgroup)_$(archType)_$(buildConfig)'

--- a/src/coreclr/build-runtime.cmd
+++ b/src/coreclr/build-runtime.cmd
@@ -42,6 +42,7 @@ set __BuildAll=
 
 set __TargetArchX64=0
 set __TargetArchX86=0
+set __TargetArchArm=0
 set __TargetArchArm64=0
 
 set __BuildTypeDebug=0
@@ -86,6 +87,7 @@ if /i "%1" == "--help" goto Usage
 if /i "%1" == "-all"                 (set __BuildAll=1&shift&goto Arg_Loop)
 if /i "%1" == "-x64"                 (set __TargetArchX64=1&shift&goto Arg_Loop)
 if /i "%1" == "-x86"                 (set __TargetArchX86=1&shift&goto Arg_Loop)
+if /i "%1" == "-arm"                 (set __TargetArchArm=1&shift&goto Arg_Loop)
 if /i "%1" == "-arm64"               (set __TargetArchArm64=1&shift&goto Arg_Loop)
 
 if /i "%1" == "-debug"               (set __BuildTypeDebug=1&shift&goto Arg_Loop)
@@ -99,6 +101,7 @@ REM don't add more, use the - syntax instead
 if /i "%1" == "all"                 (set __BuildAll=1&shift&goto Arg_Loop)
 if /i "%1" == "x64"                 (set __TargetArchX64=1&shift&goto Arg_Loop)
 if /i "%1" == "x86"                 (set __TargetArchX86=1&shift&goto Arg_Loop)
+if /i "%1" == "arm"                 (set __TargetArchArm=1&shift&goto Arg_Loop)
 if /i "%1" == "arm64"               (set __TargetArchArm64=1&shift&goto Arg_Loop)
 
 if /i "%1" == "debug"               (set __BuildTypeDebug=1&shift&goto Arg_Loop)
@@ -162,7 +165,7 @@ if defined VCINSTALLDIR (
 
 if defined __BuildAll goto BuildAll
 
-set /A __TotalSpecifiedTargetArch=__TargetArchX64 + __TargetArchX86 + __TargetArchArm64
+set /A __TotalSpecifiedTargetArch=__TargetArchX64 + __TargetArchX86 + __TargetArchArm + __TargetArchArm64
 if %__TotalSpecifiedTargetArch% GTR 1 (
     echo Error: more than one build architecture specified, but "all" not specified.
     goto Usage
@@ -173,6 +176,7 @@ if "%__ProcessorArch%"=="" set __ProcessorArch=%PROCESSOR_ARCHITECTURE%
 
 if %__TargetArchX64%==1   set __TargetArch=x64
 if %__TargetArchX86%==1   set __TargetArch=x86
+if %__TargetArchArm%==1   set __TargetArch=arm
 if %__TargetArchArm64%==1 set __TargetArch=arm64
 if "%__HostArch%" == "" set __HostArch=%__TargetArch%
 
@@ -187,6 +191,10 @@ if %__BuildTypeChecked%==1  set __BuildType=Checked
 if %__BuildTypeRelease%==1  set __BuildType=Release
 
 if %__EnforcePgo%==1 (
+    if %__TargetArchArm%==1 (
+        echo NOTICE: enforcepgo does nothing on arm architecture
+        set __EnforcePgo=0
+    )
     if %__TargetArchArm64%==1 (
         echo NOTICE: enforcepgo does nothing on arm64 architecture
         set __EnforcePgo=0
@@ -443,7 +451,7 @@ REM ============================================================================
 
 set __TargetArchList=
 
-set /A __TotalSpecifiedTargetArch=__TargetArchX64 + __TargetArchX86 + __TargetArchArm64
+set /A __TotalSpecifiedTargetArch=__TargetArchX64 + __TargetArchX86 + __TargetArchArm + __TargetArchArm64
 if %__TotalSpecifiedTargetArch% EQU 0 (
     REM Nothing specified means we want to build all architectures.
     set __TargetArchList=x64 x86 arm arm64
@@ -453,6 +461,7 @@ REM Otherwise, add all the specified architectures to the list.
 
 if %__TargetArchX64%==1      set __TargetArchList=%__TargetArchList% x64
 if %__TargetArchX86%==1      set __TargetArchList=%__TargetArchList% x86
+if %__TargetArchArm%==1      set __TargetArchList=%__TargetArchList% arm
 if %__TargetArchArm64%==1    set __TargetArchList=%__TargetArchList% arm64
 
 set __BuildTypeList=
@@ -536,7 +545,7 @@ echo All arguments are optional. The options are:
 echo.
 echo.-? -h -help --help: view this message.
 echo -all: Builds all configurations and platforms.
-echo Build architecture: one of -x64, -x86, -arm64 ^(default: -x64^).
+echo Build architecture: one of -x64, -x86, -arm, -arm64 ^(default: -x64^).
 echo Build type: one of -Debug, -Checked, -Release ^(default: -Debug^).
 echo -component ^<name^> : specify this option one or more times to limit components built to those specified.
 echo                     Allowed ^<name^>: hosts jit alljits runtime paltests iltools nativeaot spmi

